cache/models/samples.js

```diff
@@ -406,52 +424,53 @@ function upsertOneSample(sampleQueryBodyObj, isBulk, user) {
         }
 
         // sampleQueryBodyObj updated with fields
-        updateSampleAttributes(sampleQueryBodyObj, sample, aspectObj);
-
-        if (sample) { // if sample exists, just update sample
-          delete sampleQueryBodyObj.name; // to avoid updating sample name
-          logInvalidHmsetValues(sampleKey, sampleQueryBodyObj);
-          return redisClient.hmsetAsync(sampleKey, sampleQueryBodyObj);
-        }
+        return updateSampleAttributes(sampleQueryBodyObj, sample);
```

Comment:
This function is not used anymore. Can we just get rid of this function? We are using upsertOneParsedSample function instead.

Reply:
Actually, if we're no longer doing the aspect and subject lookups outside of the sample loop, we should probably go back to using this one and remove upsertOneParsedSample. (or just rename upsertOneParsedSample)

However, if we use that same pattern for the tags and status lookups as I suggested above, we would want to stick with upsertOneParsedSample and remove this one.

Reply:
I think this is good idea - Moving the tags and status lookups outside the sample loop. And if we are thinking on those lines, maybe we should move checkPermissions also out of samples loop too?
If we plan to go that route, I think that should be another story because it will involve moving critical code around. I don't want that optimization to get lost in the realtime format changes.

Reply:
Makes sense. I will create a story.